src/radical/pilot/agent/lm/jsrun.py

```diff
@@ -66,25 +70,34 @@ def _create_resource_set_file(self, slots, uid, sandbox):
 
         uid     : unit ID (string)
         sandbox : unit sandbox (string)
+        mpi     : MPI or not (bool, default: False)
 
         """
 
-        rs_id  = 0
-        rs_str = ''
-
+        rs_str = 'cpu_index_using: physical\n'
+        rank = 0
         for node in slots['nodes']:
-
-            cores = ' '.join([str(core_set[0]) for core_set
-                                               in  node['core_map']])
-            gpus  = ' '.join([str(gpu_set[0])  for gpu_set
-                                               in  node['gpu_map']])
-
-            rs_str           += 'RS %d: {'  % rs_id
-            rs_str           += ' host: %d' % node['uid']
-            if cores: rs_str += ' cpu: %s'  % cores
-            if gpus : rs_str += ' gpu: %s'  % gpus
-            rs_str           += ' }\n'
-            rs_id            += 1
+            if node['gpu_map']: 
+                node_map = zip(node['core_map'], node['gpu_map'])
```

Comment:
This assumes that both maps have the same length?  If so, is that guaranteed?  If not,  what happens to the superfulous nodes which get dropped in the zipped result?

Reply:
Hmmm you are right. I went with the assumption that we have GPUs per process. I'll fix it. Sorry

Reply:
We might have, with our recent discussion on only allowing uniform processes and all - but in the current state I am not sure the scheduler honors that, so lets stay on the save side for the time being...

Thanks!